scripts/addr2line.py

```diff
@@ -270,6 +273,12 @@ def split_addresses(self, addrstring: str, default_path: Optional[str] = None):
 
         def __call__(self, line: str) -> dict[str, Any] | None:
 
+            # quick up front check to eliminate a line from contention, which is at least
+            # 30x faster than just diving into the regex matching (for one test file)
+            if not ("0x" in line or "0X" in line or self.separator_re.match(line)):
+                # no addresses in this line, so it is not a backtrace line
+                return None
```

Comment:
Very discouraging that we have to bypass re, regular expressions are supposed to be fast.